README.md

```diff
@@ -0,0 +1,224 @@
+thread you must call `xtm_queue_notify_consumer`. The less often you notify,
+the greater the performance, but the greater the latency.
+Function accepts flags, which defined it's behaviour:
+XTM_QUEUE_PRODUCER_NEEDS_NOTIFICATIONS - if this flag is set, producer sets,
+special flag in queue internals, if push fails. Consumer can check this flag,
+using `xtm_queue_get_reset_was_full` function and notifies producer.
+Returns 0 if queue has space. Otherwise -1 with errno set to `ENOBUFS`.
+
+## xtm_queue_consumer_fd
+
+Returns file descriptor, that should be watched by consumer thread to
+become readable. When it became readable, consumer should call one of
+the consumer functions: `xtm_queue_pop_ptrs` or `xtm_queue_invoke_funs_all`.
+
+## xtm_queue_producer_fd
+
+Return file descriptor, that should be watched by producer thread to become
+readable in case when `xtm_queue_push_fun` or `xtm_queue_push_ptr` fails. When
+it became readable, producer may try push to the queue again. Very rarely,
+there may be a situation (because of race between producer and consumer
+threads), when the descriptor became readable, but queue is still full. In this
+case producer thread, need to poll this descriptor again.
+
+## xtm_queue_invoke_funs_all
+
+Function calls all functions contained in the queue at the time this function is
+called. We do not call those functions, that the producer thread can add at the
+moment, when this function is already called, in order to prevent an infinite loop
+and hunger. Also, if the queue size is very large, then hunger is still possible,
+since this function does not allow limiting the number of called functions.
+If producer thread push functions with XTM_QUEUE_PRODUCER_NEEDS_NOTIFICATIONS
+flag, user should retrieves and resets "producer failed to put an item
+in the queue and expects notification" flag, using `xtm_queue_get_reset_was_full`.
+If this flag, was true, user should notify producer thread (see simple example
+further).
+Return count of invoked functions.
+
+## xtm_queue_push_ptr
+
+Function push pointer to the queue. This function does not notify the consumer
+thread, but only push to the queue. To notify the consumer thread you must call
+`xtm_queue_notify_consumer`. The less often you notify, the greater the performance,
+but the greater the latency.
+Function accepts flags, which defined it's behaviour:
+XTM_QUEUE_PRODUCER_NEEDS_NOTIFICATIONS - if this flag is set, producer sets,
+special flag in queue internals, if push fails. Consumer can check this flag,
+using `xtm_queue_get_reset_was_full` function and notifies producer.
+Returns 0 if queue has space. Otherwise -1 with errno set to `ENOBUFS`.
+
+## xtm_queue_pop_ptrs
+
+Function gets up to count elements from the queue and save them in pointer array.
+If producer thread push ptrs with XTM_QUEUE_PRODUCER_NEEDS_NOTIFICATIONS
+flag, user should retrieves and resets "producer failed to put an item
```

Comment:
retrieves -> retrieve, resets -> reset.
You can try to use some online spell checker, or ask for gramarly licence from the doc team. Or copy-paste the text into google doc, it can do simple checks.

Reply:
Fixed